ignore/ignore.go

```diff
@@ -0,0 +1,73 @@
+// package ignore provides helpers to support ignore-files similar to .gitignore
+package ignore
+
+import (
+	"bufio"
+	"io"
+	"strings"
+)
+
+var (
+	LineComment = "#"
+	IgnoreFile  = ".sourcegraph/sourcegraphignore"
+)
+
+type IgnoreMatcher struct {
+	IgnoreList []string
+	Strip      int
+}
+
+// StripComponents removes the specified number of leading path
+// elements. Pathnames with fewer elements will return the empty string.
+func StripComponents(path string, count int) string {
+	for i := 0; path != "" && i < count; i++ {
+		i := strings.Index(path, "/")
+		if i < 0 {
+			return ""
+		}
+		path = path[i+1:]
+	}
+	return path
+}
+
+// ParseIgnoreFile parses an ignore-file according to the following rules
+//
+// - each line represents a path relative to the root of the repository
+// - lines starting with # are ignored
+// - empty lines are ignored
+// - if not present, a trailing / is implicit
+func ParseIgnoreFile(r io.Reader) (patterns []string, error error) {
```

Comment:
Why don't we make this directly return  `*IgnoreMatcher` to simplify callers?

Reply:
no reason once we remove `strip`. Before it would have been strange that the parser knows about `strip`.

Reply:
If we need to keep it the caller of ParseIgnoreFile could always set Strip afterwards.

Reply:
I removed `strip` and `ParseIgnore` is now returns `Matcher`.